### ISSUE TITLE
Overfit when using settings in fairseq finetune roberta-base with cola datset

### ISSUE THREAD

**Issue by HaoKang-Timmy** [facebookresearch/fairseq#4453]
### Before asking:
1. search the issues.
2. search the docs.

#### What is your question?
Hi, I am using settings provided in 
https://github.com/facebookresearch/fairseq/blob/main/examples/roberta/config/finetuning/cola.yaml
to finetune roberta-base with COLA dataset.
But the result get overfit after 5 epochs(total 10 epochs)
I will show you the log
```
step:0  loss_train:0.5412032071794017  acc1_train:0.7495318352059925  loss_val:0.4549379467820892  acc1_val:0.8009615384615385  time_per_batch:0.31866262914536153  matthew:{'matthews_correlation': 0.5053519875589046}
step:1  loss_train:0.3795412516514404  acc1_train:0.8403558052434457  loss_val:0.4143909378048892  acc1_val:0.8355769230769231  time_per_batch:0.3227173107840149  matthew:{'matthews_correlation': 0.5968609585806229}
step:2  loss_train:0.24929316284148936  acc1_train:0.901685393258427  loss_val:0.4335532639408484  acc1_val:0.8413461538461539  time_per_batch:0.3201296655426311  matthew:{'matthews_correlation': 0.6119568895078864}
step:3  loss_train:0.19517714867877603  acc1_train:0.9250936329588015  loss_val:0.4370046470475455  acc1_val:0.8509615384615384  time_per_batch:0.3200738952400979  matthew:{'matthews_correlation': 0.6416536225426047}
step:4  loss_train:0.13300421790870373  acc1_train:0.9536516853932584  loss_val:0.6774607508492548  acc1_val:0.8173076923076923  time_per_batch:0.3177395130364636  matthew:{'matthews_correlation': 0.5480627318306406}
step:5  loss_train:0.09997520692213108  acc1_train:0.9681647940074907  loss_val:0.6605808303078028  acc1_val:0.8221153846153846  time_per_batch:0.3189116533329424  matthew:{'matthews_correlation': 0.5611387478013941}
step:6  loss_train:0.08848711905485605  acc1_train:0.9719101123595506  loss_val:0.6310205688335163  acc1_val:0.8336538461538462  time_per_batch:0.31638000073950834  matthew:{'matthews_correlation': 0.5930335375794208}
step:7  loss_train:0.04765812121168187  acc1_train:0.9836142322097379  loss_val:0.6625456401776039  acc1_val:0.8317307692307693  time_per_batch:0.33055903402607095  matthew:{'matthews_correlation': 0.5877175973073943}
step:8  loss_train:0.06017280390435108  acc1_train:0.9808052434456929  loss_val:0.720333665547794  acc1_val:0.8269230769230769  time_per_batch:0.3190241895811388  matthew:{'matthews_correlation': 0.5746983228033765}
step:9  loss_train:0.03599870126770307  acc1_train:0.9897003745318352  loss_val:0.7288790373150662  acc1_val:0.825  time_per_batch:0.3157058311312386  matthew:{'matthews_correlation': 0.5701291603315101}
```
I changed weight decay from 0.1 to 1e-4, since 0.1 can not be tuned to an acceptable accuracy.
Why validation accuracy gets really bad after 5 epochs?
#### What have you tried?
I have changed different settings in huggingface and fairseq
#### What's your environment?

-fairseq 1.0
-pytorch 1.11.0
